examples/gno.land/r/sys/users/users.gno

```diff
@@ -0,0 +1,52 @@
+package users
+
+import (
+	"std"
+
+	"gno.land/r/demo/users"
+)
+
+// For now this package is disable by default
+var enable = false
+
+var admin = "g1us8428u2a5satrlxzagqqa5m6vmuze025anjlj"
+
+// Admin call to Activate or Desactivate this package.
+func AdminEnable() {
+	assertIsAdmin()
+
+	enable = true
+}
+
+func AdminDisable() {
+	assertIsAdmin()
+
+	enable = false
+}
+
+func IsEnable() bool { return enable }
+
+// This method ensure that the given address have ownership on the given name.
+func IsAuthorizedAddressForName(address std.Address, name string) bool {
+	if !enable {
+		return true
+	}
+
+	// Allow user with their own address as name
+	if address.String() == name {
+		return true
+	}
+
+	if user := users.GetUserByName(name); user != nil {
```

Comment:
I would love to be able to have multiple names tied to a single addresses, because the current system requires me to track which addresses (keys) I can use for which namespaces :/

Reply:
i've update the package so the checkFunc is callback that can be updated later by the owner and add a way to support this somehow, wdyt ?

Reply:
I love it, let's roll with this approach for now